src/Mempool.h

```diff
@@ -136,11 +136,15 @@ struct Mempool
     using NewTxsMap = std::unordered_map<TxHash, std::pair<Mempool::TxRef, bitcoin::CTransactionRef>, HashHasher>;
     /// The scriptHashes that were affected by this refresh/synch cycle. Used for notifications.
     using ScriptHashesAffectedSet = std::unordered_set<HashX, HashHasher>;
+    /// The scriptHash transactions that were affected by this refresh/synch cycle. Used for notifications.
+    using ScriptHashTransactionsAffectedMap = std::unordered_map<HashX, std::set<TxHash>, HashHasher>;
```

Comment:
I use ordered set, so that the hash of the set is the same for the same transaction composition. Also set comparison will then be trivial